### PR TITLE
Fix problems found while publishing 7.21.0

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

I had to manually release 7.21.0 because these problems made CI fail (https://github.com/babel/babel/actions/runs/4224580190/jobs/7335726019)

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15440"></a>

### CODE REVIEW

File: packages/babel-parser/typings/babel-parser.d.ts [hunk 1]
```diff
@@ -230,4 +230,4 @@ type ParseResult<Result> = Result & {
   errors: ParseError[];
 };
 
-export { DecoratorsPluginOptions, FlowPluginOptions, ParseError, ParserOptions, ParserPlugin, ParserPluginWithOptions, PipelineOperatorPluginOptions, RecordAndTuplePluginOptions, TypeScriptPluginOptions, parse, parseExpression, tokTypes };
+export { DecoratorsPluginOptions, FlowPluginOptions, ParseError, ParseResult, ParserOptions, ParserPlugin, ParserPluginWithOptions, PipelineOperatorPluginOptions, RecordAndTuplePluginOptions, TypeScriptPluginOptions, parse, parseExpression, tokTypes };
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/typings/babel-parser.d.ts:233]: This is a regression, for some reason the file published in 7.21.0 didn't include `ParseResult`.

**Comment by liuxingbaoyu** [comment]
These look like warnings, and I think the real failure reason has to do with `FORCE_COLOR`.

**Comment by liuxingbaoyu** [comment]
https://github.com/babel/babel/pull/14572/files#diff-10d032f00c2fe160bd8376108cca5a4c66d73998c3a47008f89e58eae5ea1356
Modifications to this file may be able to fix this problem.
image

File: packages/babel-parser/typings/babel-parser.source.d.ts [hunk 2]
```diff
@@ -40,6 +40,12 @@ export interface ParserOptions {
    */
   allowReturnOutsideFunction?: boolean;
 
+  /**
+   * By default, new.target use is not allowed outside of a function or class.
+   * Set this to true to accept such code.
+   */
+  allowNewTargetOutsideFunction?: boolean;
```
Comment (liuxingbaoyu) [inline packages/babel-parser/typings/babel-parser.source.d.ts:47]: Good catch! Let's add `annexB` together.

**Review by JLHwung — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
I'm trying to reproduce the original failure locally.

**Comment by nicolo-ribaudo** [comment]
I tried `make prepublish | cat` on main but I still don't see the error :confused: 
I hope this PR fixes it, but the only way to know is to try to release a patch (since the error doesn't happen in our e2e publish tests)

**Comment by liuxingbaoyu** [comment]
Whoops, I just remembered. Probably because github's terminal doesn't fully support colors, I remember this was fixed in newer dependencies, but we don't seem to be updating it because of babel8.

**Comment by nicolo-ribaudo** [comment]
Ohh I got it.
- On normal CI, we run Jest with `--color`
- On e2e publish, we run `prepublish-build` instead of `prepublish`, and it doesn't run tests

I'll add back `FORCE_COLOR` in the release GH workflow.

**Comment by liuxingbaoyu** [comment]
Also I found that `babel-code-frame` and `babel-highlight` both throw exceptions.

**Review by liuxingbaoyu — APPROVED** [review]

**Pull request merged**